lambda/lambda.mli

```diff
@@ -1040,3 +1064,13 @@ val simple_prim_on_values
 -> arity:int
 -> alloc:bool
 -> external_call_description
+
+(* Info for a compilation unit that implements a parameter (i.e., is an argument
+   for that parameter) *)
+
+type arg_descr =
```

Comment:
It's not just the argument to `-as-argument-for`, though (that would be the `arg_param` field, which I think is a more helpful name; I'll add a comment about the connection)

Reply:
Oh, leaving it as a non-documentation comment since it would be the only doc comment around, which only raises questions

Reply:
> It's not just the argument to -as-argument-for, though

Do you mean there is an extra field `arg_block_field`? But that's like an implementation-level details about "in what way is this module an argument for `arg_param`".

Reply:
Right. But that's why I hesitate to name the whole thing after the CLI flag: it _isn't_ just the value of the flag.

Reply:
I think it's fine for it to contain some middle-end details. `Lambda.lfunction` has plenty of such things, and we still call it function, because the correspondance to front-end is still useful.